library/ceph_disks.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/python
+#coding: utf-8 -*-
+
+DOCUMENTATION = '''
+---
+author: Siva Mullapudi
+module: ceph_disks
+short_description: Outputs a list of valid devices
+description:
+    - Outputs a list of devices based on the osd type.
+options: none
+'''
+
+EXAMPLES = '''
+# Assuming a journal_collocation osd type
+- ceph_disks:
+  register: ssds
```

Comment:
If this were a proper fact module, there would be no need to `register` the results. The facts gathered would automatically be added to the host variables.